packages/grid/src/GridRenderer.js

```diff
@@ -1142,16 +1155,67 @@ class GridRenderer {
     context.fillStyle = headerBackgroundColor;
     context.fillRect(0, 0, width, columnHeaderHeight);
 
-    // Draw the separators
+    context.fillStyle = headerColor;
+
+    // Visible columns.
+    for (let i = 0; i < visibleColumns.length; i += 1) {
+      const column = visibleColumns[i];
+      const columnWidth = visibleColumnWidths.get(column);
+      const x = visibleColumnXs.get(column) + gridX;
+      this.drawColumnHeader(context, state, column, x, columnWidth);
+    }
+
+    if (containsFrozenColumns) {
+      floatingLeftColumnsWidth =
+        visibleColumnXs.get(floatingLeftColumnCount - 1) +
+        visibleColumnWidths.get(floatingLeftColumnCount - 1);
+
+      // Frozen columns' background
+      context.fillStyle = headerBackgroundColor;
+      context.fillRect(0, 0, floatingLeftColumnsWidth, columnHeaderHeight);
```

Comment:
As mentioned in my other comment, this is drawing the column background in the incorrect location if the row headers are visible. You can test this using the style guide, go to http://localhost:4000/styleguide... Need to set `floatingLeftColumnCount` in MockGridModel to something other than 0 as well temporarily.

```suggestion
      context.fillRect(gridX, 0, floatingLeftColumnsWidth, columnHeaderHeight);
```